src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -48,9 +48,11 @@ public class VsPackageInstaller : IVsPackageInstaller2
             _sourceRepositoryProvider = sourceRepositoryProvider;
             _settings = settings;
             _solutionManager = solutionManager;
-            _projectContext = new VSAPIProjectContext();
             _packageServices = packageServices;
             _deleteOnRestartManager = deleteOnRestartManager;
+
+            _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
```

Comment:
here `VSAPIProjectContext` is being initialized in bg thread via `PumpingJTF.Run` and during it's initialization, it tries to get `ISourceControlManagerProvider` service via `ServiceLocator.GetInstanceSafe` which is where it will try to go to UI thread via another `JTF.Run` to get the service. 

So `VSAPIProjectContext` is never constructed on UI thread so I'm not able to understand why should we move it to UI thread or add a check for it?